### PR TITLE
Job merging state

### LINKED ISSUE #3223: Move egressing into merge process (Add a pipeline state 'egressing').
Currently, if a pipeline has `egress` set, then after a job finishes running it appears to get into a funny state: the job's output commit gets finished, but the job stays in state `running` until egress is finished. This is confusing to users and also potentially lossy, if the master dies while egress is running (which can happen because egress is slow and potentially consumes a lot of memory).

It would be better if we indicated this transition in the JobInfo, so that a) it can be displayed to users, and b) the worker master could detect it and resume egress on failure

### PR DESCRIPTION
**Pull request by ysimonson**
Fixes #3223

### CODE REVIEW

File: src/server/pachyderm_test.go [hunk 1]
```diff
@@ -2138,8 +2138,11 @@ func TestUpdatePipelineRunningJob(t *testing.T) {
 		if len(jobInfos) != 1 {
 			return fmt.Errorf("wrong number of jobs")
 		}
-		if pps.JobState_JOB_RUNNING != jobInfos[0].State {
-			return fmt.Errorf("wrong state: %v for %s", jobInfos[0].State, jobInfos[0].Job.ID)
+
+		state := jobInfos[0].State
+
+		if state != pps.JobState_JOB_RUNNING && state != pps.JobState_JOB_MERGING {
+			return fmt.Errorf("wrong state: %v for %s", state, jobInfos[0].Job.ID)
```
Comment (jdoliner) [inline src/server/pachyderm_test.go]: This doesn't guarantee that the job enters state merging, but I think that'd be pretty annoying to do in a test and would be flaky, so I'm cool with this.

**Comment by ysimonson** [comment]
@jdoliner same tests are failing even with the performance improvements merged in. Given that, I think these changes are producing some sort of (non-deterministic) deadlock state, but I'm not sure how.

**Comment by jdoliner** [comment]
Yeah, these might actually not be flakes. I'd try grepping for everywhere in the codebase we use the running state and see if we should be using merging there too.

**Comment by ysimonson** [comment]
@jdoliner anything I should wait for, or good to merge at this point?

**Comment by jdoliner** [comment]
@ysimonson nope, that big green button wants you to hit it :)

**Pull request merged**